File: Sum_Difference_Product.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 15 00:09:33 2022

@author: bushra

------------------------------------------------------------------------------
   Name:     BI_IS221_PP_2_3
   Author:   Bushra Ibrahim
   Date:     Apr 15, 2022
   Language: Python
   Purpose:  The purpose of this program is to take two floating point numbers 
             you enter and print their sum, difference, and product.
------------------------------------------------------------------------------
   ChangeLog:
   
   Who:      Bushra Ibrahim           
   Date:     Apr 15, 2022
   Desc.:    This is the original version of the code.
------------------------------------------------------------------------------

"""

# 2.3
# Write a program that reads two floating point numbers and prints their sum, difference, and product.

def BI_IS221_PP_2_3():
    
    print("\nThis program takes two floating point numbers you enter and prints their sum, difference, and product.\n")
    
    blFlt1 = False
    blFlt2 = False
    
    while blFlt1 is False:
        fltOne = input("Please enter your first float: ")
        try:
            i = float(fltOne)
            if type(i) == float:
                blFlt1 = True
                break
        except ValueError:
            print("That was not a float, please try again.")
    
    while blFlt2 is False:
        fltTwo = input("Please enter your second float: ")
        try:
            i = float(fltTwo)
            if type(i) == float:
                blFlt2 = True
                break
        except ValueError:
            print("That was not a float, please try again.")
            
    fltSum = float(fltOne) + float(fltTwo)
    fltDiff = float(fltOne) - float(fltTwo)
    fltProd = float(fltOne) * float(fltTwo)
    
    print("\nThe sum of your two floats is: ", fltSum)
    print("The difference of your two floats is: ", fltDiff)
    print("The product of your two floats is: ", fltProd)
    
BI_IS221_PP_2_3()

File: Sum_Difference_Product.java
/*
------------------------------------------------------------------------------
   Name:     Sum_Difference_Product
   Author:   Bushra Ibrahim
   Date:     Feb 9, 2022
   Language: Java
   Purpose:  The purpose of this program is to take 2 floating point variable
             values from the user (which will look just like inputting integers)
             and returning their sum, difference, and product. (see PP 2.3)
------------------------------------------------------------------------------
   ChangeLog:
   Who:      Bushra Ibrahim           Date:     Feb 9, 2022
   Desc.:    This is the original version of the code.
------------------------------------------------------------------------------
*/

package sum_difference_product;

import java.util.Scanner;

public class Sum_Difference_Product {


    public static void main(String[] args) {
        Scanner scnKey = new Scanner (System.in);
        
        System.out.println("Hello. This program takes 2 numbers from you" +
                          " and prints back their sum, difference, and product" +
                          " as floating point variables. Please enter your" +
                          " first number.");
        float fltOne = scnKey.nextInt();
        scnKey.nextLine();
        System.out.println("Please enter your second number.");
        float fltTwo = scnKey.nextInt();
        scnKey.nextLine();
        System.out.println("Thank you. Here are the sum, difference, and"
                +          " product of the 2 numbers you entered.");
        float fltSum = (fltOne + fltTwo);
        System.out.println("Sum: " + fltSum);
        float fltDiff = (fltOne - fltTwo);
        System.out.println("Difference: " + fltDiff);
        float fltProd = (fltOne * fltTwo);
        System.out.println("Product: " + fltProd);
    }

}
